component/src/main/java/io/siddhi/extension/io/grpc/sink/GrpcSink.java

```diff
@@ -387,8 +362,30 @@ public void connect() throws ConnectionUnavailableException {
             this.channel = managedChannelBuilder.build();
             if (serviceConfigs.isDefaultService()) {
                 this.asyncStub = EventServiceGrpc.newStub(channel);
+                if (metadataOption != null) {
+                    if (metadataOption.isStatic()) {
+                        asyncStub = attachMetaDataToStub(null, asyncStub);
+                    }
+                }
+                requestObserver = ((EventServiceGrpc.EventServiceStub) asyncStub).consume(responseObserver);
             } else {
+                rpcMethod = getRpcMethod(serviceConfigs, siddhiAppName, streamID);
                 this.asyncStub = createStub(serviceConfigs);
+                if (metadataOption != null) {
+                    if (metadataOption.isStatic()) {
+                        asyncStub = attachMetaDataToStub(null, asyncStub);
+                    }
+                }
+                try {
+                    if (rpcMethod.getParameterCount() == 1) {
+                        requestObserver = (StreamObserver) rpcMethod.invoke(asyncStub, responseObserver);
+                    }
+                } catch (IllegalAccessException | InvocationTargetException e) { //throws from 'invoke'
+                    throw new SiddhiAppValidationException(siddhiAppName + ":" + streamID + ": Invalid method name " +
```

Comment:
Don't throw Appvalidation exception at connect level. It should be a runtime exception